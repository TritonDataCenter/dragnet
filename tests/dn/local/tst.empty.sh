#!/bin/bash

#
# tst.empty.sh: tests basic "scan" operations on empty files
#

set -o errexit
. $(dirname $0)/../common.sh

tmpfile="/var/tmp/$(basename $0).$$"
echo "using tmpfile \"$tmpfile\"" >&2

function scan
{
	echo "# dn scan" "$@"
	dn scan "$@" devnull 2>&1
	echo

	echo "# dn scan --points" "$@"
	dn scan --points "$@" devnull 2>&1 | sort -d
	echo
}

function query
{
	echo "# dn query" "$@"
	dn query --interval=all "$@" devnull 2>&1
}

dn_clear_config
dn datasource-add devnull --path=/dev/null --index-path=$tmpfile
scan --counters
scan -b timestamp
scan -b timestamp[aggr=quantize]
scan -b timestamp[aggr=quantize],req.method
scan -f '{ "eq": [ "audit", true ] }' -b timestamp[aggr=quantize],req.method
scan --counters -f '{ "eq": [ "audit", true ] }'

echo "creating index" >&2
dn metric-add devnull total
dn build --interval=all devnull
query --counters

echo "creating index" >&2
dn metric-add devnull met -b req.method,latency[aggr=quantize]
dn build --interval=all devnull
query --counters
query -f '{ "eq": [ "req.method", "GET" ] }'
query -b req.method
query -b latency
query --counters -b latency
dn_clear_config
rm -rf $tmpfile
