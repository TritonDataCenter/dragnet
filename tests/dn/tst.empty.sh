#!/bin/bash

#
# tst.empty.sh: tests basic "scan" operations on empty files
#

set -o errexit
. $(dirname $0)/common.sh

tmpfile="/var/tmp/$(basename $0).$$"
echo "using tmpfile \"$tmpfile\"" >&2

function scan-file
{
	echo "# dn scan-file" "$@"
	dn scan-file "$@" /dev/null 2>&1
	echo

	echo "# dn scan-file --points" "$@"
	dn scan-file --points "$@" /dev/null 2>&1 | sort -d
	echo
}

function query
{
	echo "# dn query-file" "$@"
	dn query-file "$@" $tmpfile 2>&1
}

scan-file --counters
scan-file -b timestamp
scan-file -b timestamp[aggr=quantize]
scan-file -b timestamp[aggr=quantize],req.method
scan-file -f '{ "eq": [ "audit", true ] }' -b timestamp[aggr=quantize],req.method
scan-file --counters -f '{ "eq": [ "audit", true ] }'

echo "creating index" >&2
dn index-file /dev/null $tmpfile
query --counters

echo "creating index" >&2
dn index-file -c req.method,latency[aggr=quantize] /dev/null $tmpfile
query --counters
query -f '{ "eq": [ "req.method", "GET" ] }'
query -b req.method
query -b latency
query --counters -b latency
