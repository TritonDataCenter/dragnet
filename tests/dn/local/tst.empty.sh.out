# dn scan --counters
VALUE
    0
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2
FindFeedback       nchrdevs:           1
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dn scan --points --counters
Aggregator         noutputs:           1
{"fields":{},"value":0}
FindFeedback       nchrdevs:           1
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2

# dn scan -b timestamp

# dn scan --points -b timestamp

# dn scan -b timestamp[aggr=quantize]

# dn scan --points -b timestamp[aggr=quantize]

# dn scan -b timestamp[aggr=quantize],req.method

# dn scan --points -b timestamp[aggr=quantize],req.method

# dn scan -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dn scan --points -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dn scan --counters -f { "eq": [ "audit", true ] }
VALUE
    0
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2
FindFeedback       nchrdevs:           1
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dn scan --points --counters -f { "eq": [ "audit", true ] }
Aggregator         noutputs:           1
{"fields":{},"value":0}
FindFeedback       nchrdevs:           1
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2

# dn query --counters
VALUE
    0
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
FindFeedback       nregfiles:          1
Index List         ninputs:            1
Index List         noutputs:           1
Index Result Aggregator ninputs:            1
Index Result Aggregator noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dn query --counters
VALUE
    0
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
FindFeedback       nregfiles:          1
Index List         ninputs:            1
Index List         noutputs:           1
Index Result Aggregator ninputs:            1
Index Result Aggregator noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dn query -f { "eq": [ "req.method", "GET" ] }
VALUE
    0
# dn query -b req.method
# dn query -b latency
# dn query --counters -b latency
FindStart          ninputs:            1
FindStart          noutputs:           1
FindStatter        ninputs:            2
FindStatter        noutputs:           2
FindTraverser      ninputs:            2
FindTraverser      noutputs:           2
FindFeedback       ninputs:            2
FindFeedback       noutputs:           1
FindFeedback       nregfiles:          1
Flattener          noutputs:           1
